tests/gocase/unit/keyspace/keyspace_test.go

```diff
@@ -77,6 +74,37 @@ func TestKeyspace(t *testing.T) {
 		require.Equal(t, []string{"foo_a", "foo_b", "foo_c", "key_x", "key_y", "key_z"}, keys)
 	})
 
+	t.Run("KEYS with non-trivial patterns", func(t *testing.T) {
+		require.NoError(t, rdb.FlushDB(ctx).Err())
+		for _, key := range []string{"aa", "aab", "aabb", "ab", "abb"} {
+			require.NoError(t, rdb.Set(ctx, key, "hello", 0).Err())
+		}
+
+		keys := rdb.Keys(ctx, "a*").Val()
```

Comment:
Could you add a case to ensure that `aa` just matches `aa`, and `aa*` can match `aa`, `aab`, ..?